### ISSUE TITLE
Experimental support for decorators is a feature that is subject to change ...

### ISSUE THREAD

**Issue by msegmx** [microsoft/vscode#102874]
- VSCode / OS Version:
```
Version: 1.47.1
Commit: 485c41f9460bdb830c4da12c102daff275415b53
Date: 2020-07-14T00:13:57.513Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.4.0-40-generic
```

Steps to Reproduce:

1. create a new .ts file and paste this code in it:

```
import { Injectable } from '@angular/core';

@Injectable({
    providedIn: 'root'
})
export class SyncService {
}
```

A warning message appears: 
```
Experimental support for decorators is a feature that is subject to change in a future release. Set the 'experimentalDecorators' option in your 'tsconfig' or 'jsconfig' to remove this warning.
```

The warning message shouldn't appear because of:

```
@Injectable({
    providedIn: 'root'
})
```

Meaning this class (SyncService) doesn't have to be referenced anywhere else in the project to run as a service. Please note, there are other (earlier created) services in the project which doesn't get this warning.

2. 

![here](https://i.stack.imgur.com/MSG2V.gif)

Does this issue occur when all extensions are disabled?: Yes

**Comment by IllusionMH** [comment]
@msegmx can you show what is in your `"files"`, `"include"`, `"exclude"` of your configuration and if this file matches anything *before* imported to other file.

TL;DR; If file is not part of `"include"` and wasn't imported to any file that is in imports tree form files in `"files"` and `"include"` - it will be part of "Inferred project" that has default configuration (without decorators enabled).
You need to make sure that files you are going to works will be matched by your `"include"` configuration to make sure that they are part of proper project.

A bit more details in [similar answer](https://github.com/microsoft/TypeScript/issues/39488#issuecomment-658201133)

**Comment by msegmx** [comment]
I created the project recently with "ng new" (Angular 10).
Here are the relevant tsconfig files (the only change is "baseUrl" : "./" -> "./src"):

tsconfig.json:
```
{
  "files": [],
  "references": [
    {
      "path": "./tsconfig.app.json"
    },
    {
      "path": "./tsconfig.spec.json"
    },
    {
      "path": "./e2e/tsconfig.json"
    }
]
}
```

tsconfig.app.json:
```
{
  "extends": "./tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./out-tsc/app",
    "types": []
  },
  "files": [
    "src/main.ts",
    "src/polyfills.ts"
  ],
  "include": [
    "src/**/*.d.ts"
  ]
}
```

tsconfig.base.json:
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./src",
    "outDir": "./dist/out-tsc",
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2015",
    "module": "es2020",
    "lib": [
      "es2018",
      "dom"
    ]
  },
  "angularCompilerOptions": {
    "strictInjectionParameters": true,
    "strictTemplates": true
  }
}

```

**Comment by IllusionMH** [comment]
Yep, most likely that's the same problem described in that comment, but different observed result (I think you should have problem with auto-import of this file too).

This issue has nothing to do with VS Code and most likely is not TS issue (because it's works as expected now - only files from `files` and `include` are part of the project as documented). There are couple of opened tracking issues in Angular repo.

At this point proper solution mentioned in doc linked in https://github.com/microsoft/TypeScript/issues/39632 in section [1. Unreferenced files don’t inherit compiler options](https://docs.google.com/document/d/1eB6cGCG_2ircfS5GzpDC9dBgikeYYcMxghVH5sDESHw/edit#heading=h.telvo0tzo7t0)

**Comment by msegmx** [comment]
Indeed I have had problems with auto-completion.

Thank you so much for shedding light on this issue. Turns out every Angular developer using v10 should be aware of these changes that come with the "solutions style tsconfig" approach.

I prefer not to change the tsconfig files, but rather will continue to temporarily import the new file (service) until I actually import and use it in a component, etc. But at least I know now why this warning happens.
